dsl/src/main/scala/org/http4s/dsl/impl/Path.scala

```diff
@@ -186,7 +186,7 @@ protected class PathVar[A](cast: String => Option[A]) {
 object IntVar
     extends PathVar((str: String) =>
       try
-        Option(str.toInt)
+        Some(str.toInt)
```

Comment:
str.toIntOption ? ( might not be possible due to 2.12 )

Reply:
Just a note that `toIntOption` was added to the latest [scala-collection-compat](https://github.com/scala/scala-collection-compat/releases/tag/v2.9.0)
However, we're currently not using scala-collection-compat, we've thought about it before, but ultimately went with a lighter approach (https://github.com/http4s/http4s/issues/3945). We could presumably revisit that though.

Reply:
scala-collection-compat gets privileged access to the underlying implementation by being in the `scala` package.  Bummer.  That could be cleaned up on main, at least.

I don't know whether Scala's StringParsers is faster than Java's.  Probably yes, because it doesn't have to create an exception.  If anyone _really_ cares, there could be a 2.12 and 2.13+ implementation.